source/btn.cpp

```diff
@@ -33,10 +36,26 @@ static void btn_onpress(void)
     pc.printf(" done\n\r");
 }
 
+static void btn_set_fall(void)
+{
+    btn.fall(&btn_onpress);
+}
+
 void btn_init(void)
 {
     /* configure pushbutton */
     btn.mode(MAIN_BTN_PUPD);
-    wait(.01);
-    btn.fall(&btn_onpress);
+
+    /* Register the button fall handler after a short delay. */
+    minar::Scheduler::postCallback(FunctionPointer0<void>(btn_set_fall).bind())
+        .delay(minar::milliseconds(10))
+        .tolerance(minar::milliseconds(100));
```

Comment:
I am convinced it's broken that way: I expect that minar skips the delay altogether, as 10ms delay +/- 100ms results in 0ms delay in most cases. Either remove the tolerance or set it to something meaningful like 10% of the delay.